tests/unit/config/config_test.py

```diff
@@ -1765,22 +1767,23 @@ def test_config_valid_environment_dict_key_contains_dashes(self):
         assert services[0]['environment']['SPRING_JPA_HIBERNATE_DDL-AUTO'] == 'none'
 
     def test_load_yaml_with_yaml_error(self):
-        tmpdir = py.test.ensuretemp('invalid_yaml_test')
-        self.addCleanup(tmpdir.remove)
-        invalid_yaml_file = tmpdir.join('docker-compose.yml')
-        invalid_yaml_file.write("""
-            web:
-              this is bogus: ok: what
-        """)
+        tmpdir = tempfile.mkdtemp('invalid_yaml_test')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        invalid_yaml_file = os.path.join(tmpdir, 'docker-compose.yml')
+        with open(invalid_yaml_file, mode="w") as invalid_yaml_file_fh:
+            invalid_yaml_file_fh.write("""
+                web:
```

Comment:
Could you please make the content of the file more deterministic by aligning that on left, so that the indentation doesn't change the `assert` in line 1781? Maybe remove the first `\n` also, so that the file doesn't start with a new line.
I know that was there before, but since we are changing...

Reply:
Fixed. I'd keep the first newline there because removing it might visually hide that the file has more than one line.
```
            invalid_yaml_file_fh.write("""web:
    this is bogus: ok: what
            """)
```